include/tunable_reduction.h

```diff
@@ -357,14 +358,19 @@ namespace quda
     }
 
   public:
-    TunableMultiReduction(const LatticeField &field, int n_batch,
+    TunableMultiReduction(const LatticeField &field, unsigned int n_batch, unsigned int n_batch_block_max = 1u,
                           QudaFieldLocation location = QUDA_INVALID_FIELD_LOCATION) :
-      TunableReduction2D<block_size_y>(field, location), n_batch(n_batch)
+      TunableReduction2D<block_size_y>(field, location),
+      n_batch(n_batch),
+      n_batch_block_max(n_batch_block_max)
     {
     }
```

Comment:
I would suggest having a check of `n_batch_block_max == Arg::max_n_batch_block` here since otherwise we might get mysterious shared memory out of bound.

Reply:
I don't think that's possible at this point, since the `Arg` isn't visible here, and may not even be instantiated yet.

Reply:
Ah, but I can add that check to the launch function.  Will do that now.

Reply:
I put it in the wrong place but I think you got my point :)

Reply:
1650dc622e8d9ce45dc5c2eef6bba0f5a7b601be applies this check.  Instead of checking for equality, I instead check that the max used by the tuner does not exceed the max supported by the kernel.  There may be some run-time heuristic for example for pruning the max size, so we shouldn't insist on equality.

Reply:
Yep, I was about to suggest `<` instead of `=`.